### PR TITLE
Bump transformers version to 5.2.0

### PR DESCRIPTION
**Pull request by huydhn**
Take over the Dependabot PR from https://github.com/pytorch/pytorch/pull/175147 to fix the failures there

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @kadeng @chauhang @amjames @Lucaskabela @jataylo

### CODE REVIEW

**Comment by huydhn** [comment]
The `test/distributed/test_dynamo_distributed.py::TestFakeDistributedSingleProc::test_hf_bert_ddp_aot_eager` failure is legit. Trying to investigate it with Claude.

Transformers v5's create_bidirectional_mask produces different mask values under torch.compile vs eager execution, causing SDPA to select different CUDA kernels with different dropout RNG consumption, leading to numerically divergent outputs.

The issue is in transformers/masking_utils.py in _ignore_bidirectional_mask_sdpa:

```
  def _ignore_bidirectional_mask_sdpa(padding_mask):
      if not is_tracing(padding_mask) and (padding_mask is None or padding_mask.all()):
          return True  # skip mask → return None
      return False
```

  - Eager: is_tracing() is False → mask creation is skipped → SDPA gets attn_mask=None → FlashAttention kernel
  - Compiled: is_tracing() is True → all-True boolean mask is materialized → SDPA gets attn_mask=<bool tensor> → different kernel

The different kernels consume CUDA RNG differently for dropout, producing different outputs (max diff ~3.0 in logits for a 12-layer BERT).

This affects any non-decoder model using SDPA (the default) with dropout during training when comparing eager vs torch.compile outputs.

Reproducible snippet

```
  """
  Transformers v5: create_bidirectional_mask causes eager vs torch.compile divergence
  with SDPA attention + dropout on CUDA.

  Tested with: transformers==5.0.0 / 5.2.0, torch nightly (2.8.0.dev)
  """
  import torch
  import torch._dynamo
  from transformers import BertConfig
  from transformers.models.bert.modeling_bert import BertModel


  def reset_rng():
      torch.manual_seed(42)
      torch.cuda.manual_seed(42)


  device = "cuda:0"
  input_ids = torch.randint(0, 30522, (4, 512), device=device)

  # ---- FAILS: sdpa (default) in train mode ----
  print("=== SDPA attention (default), train mode ===")
  torch._dynamo.reset()
  config = BertConfig(attn_implementation="sdpa")
  model = BertModel(config).to(device).train()

  reset_rng()
  eager_out = model(input_ids, return_dict=True)

  reset_rng()
  compiled = torch.compile(model, backend="aot_eager")
  compiled_out = compiled(input_ids, return_dict=True)

  diff = (eager_out.last_hidden_state - compiled_out.last_hidden_state).abs().max().item()
  print(f"  max diff: {diff}")  # ~3.0 (FAIL)

  # ---- PASSES: eager attention in train mode ----
  print("\n=== Eager attention, train mode ===")
  torch._dynamo.reset()
  config2 = BertConfig(attn_implementation="eager")
  model2 = BertModel(config2).to(device).train()

  reset_rng()
  eager_out2 = model2(input_ids, return_dict=True)

  reset_rng()
  compiled2 = torch.compile(model2, backend="aot_eager")
  compiled_out2 = compiled2(input_ids, return_dict=True)

  diff2 = (eager_out2.last_hidden_state - compiled_out2.last_hidden_state).abs().max().item()
  print(f"  max diff: {diff2}")  # 0.0 (PASS)

  # ---- PASSES: sdpa but eval mode (no dropout) ----
  print("\n=== SDPA attention, eval mode (no dropout) ===")
  torch._dynamo.reset()
  config3 = BertConfig(attn_implementation="sdpa")
  model3 = BertModel(config3).to(device).eval()

  with torch.no_grad():
      eager_out3 = model3(input_ids, return_dict=True)
      compiled3 = torch.compile(model3, backend="aot_eager")
      compiled_out3 = compiled3(input_ids, return_dict=True)

  diff3 = (eager_out3.last_hidden_state - compiled_out3.last_hidden_state).abs().max().item()
  print(f"  max diff: {diff3}")  # ~7e-6 (PASS)

  # ---- PASSES: sdpa, train mode, but dropout=0 ----
  print("\n=== SDPA attention, train mode, dropout=0 ===")
  torch._dynamo.reset()
  config4 = BertConfig(attn_implementation="sdpa", hidden_dropout_prob=0.0, attention_probs_dropout_prob=0.0)
  model4 = BertModel(config4).to(device).train()

  eager_out4 = model4(input_ids, return_dict=True)
  compiled4 = torch.compile(model4, backend="aot_eager")
  compiled_out4 = compiled4(input_ids, return_dict=True)

  diff4 = (eager_out4.last_hidden_state - compiled_out4.last_hidden_state).abs().max().item()
  print(f"  max diff: {diff4}")  # ~7e-6 (PASS)
```

Expected output:
```
  === SDPA attention (default), train mode ===
    max diff: 0.0
  === Eager attention, train mode ===
    max diff: 0.0
  === SDPA attention, eval mode (no dropout) ===
    max diff: ~7e-6
  === SDPA attention, train mode, dropout=0 ===
    max diff: ~7e-6
```

Actual output:
```
  === SDPA attention (default), train mode ===
    max diff: 3.3  ← WRONG
  === Eager attention, train mode ===
    max diff: 0.0
  === SDPA attention, eval mode (no dropout) ===
    max diff: ~7e-6
  === SDPA attention, train mode, dropout=0 ===
    max diff: ~7e-6
```

Root cause: _ignore_bidirectional_mask_sdpa in masking_utils.py:263 returns True (skip mask) in eager but False (materialize mask) under torch.compile due to the is_tracing() guard. The materialized all-True boolean mask causes SDPA to select a different CUDA kernel than when attn_mask=None, and the different
  kernel consumes the CUDA RNG differently for dropout.

**Review by xmfan — APPROVED** [review]
lgtm, please notify the pt2 oncall chat before landing

**Review by malfet — APPROVED** [review]
To many `pass` to `fail_accuracy` changes, but may be this is fine

**Comment by huydhn** [comment]
Btw, this is now blocking vLLM pinned commit update. There are some models on their test suite only works with transformers 5.x now https://github.com/vllm-project/vllm/issues/34642
